### PR TITLE
Support a text wrapping preference in the textview

### PR DESCRIPTION
**Pull request by ben-p-commits**
# Background
This addresses bug #84 , and fulfills some of the requirements of #66 (the rest of which will be implemented in **CodeEdit** in a nascent PR). 

## Technical Details

Two things to note:

1. This will likely regress highlighting behavior- more detailed discussion available on [this discord channel](https://discord.com/channels/951544472238444645/952640818068463667/1059969085695340595), but TLDR: @thecoolwinter has plans to obviate the need for the removed `NSLayoutManager` code when he migrates that work to TextKit 2.

2. I went ahead and removed the invocation of STTextView's static scrollableTextView() helper function. It sets a number of properties which are immediately overwritten by our code in `loadView()`. It only obfuscates what's really happening here.

### CODE REVIEW

File: Sources/CodeEditTextView/STTextViewController.swift [hunk 1]
```diff
@@ -78,22 +83,19 @@ public class STTextViewController: NSViewController, STTextViewDelegate, ThemeAt
...
         scrollView.hasVerticalScroller = true
+        scrollView.documentView = textView
 
         rulerView = STLineNumberRulerView(textView: textView, scrollView: scrollView)
         rulerView.backgroundColor = theme.background
         rulerView.textColor = .systemGray
         rulerView.drawSeparator = false
         rulerView.baselineOffset = baselineOffset
-        rulerView.font = NSFont.monospacedDigitSystemFont(ofSize: 9.5, weight: .regular)
+        rulerView.font = NSFont(descriptor: NSFont.monospacedDigitSystemFont(ofSize: 11, weight: .regular).fontDescriptor, textTransform: AffineTransform(scaleByX: 1, byY: 1.1)) ?? .monospacedDigitSystemFont(ofSize: 11, weight: .regular)
```
Comment (lukepistrol) [inline Sources/CodeEditTextView/STTextViewController.swift:98]: Why do we need to scale the font on the y axis?

Comment (ben-p-commits) [reply Sources/CodeEditTextView/STTextViewController.swift:98]: That was mistakenly committed. in the process of cleaning this draft up- woops

Comment (ben-p-commits) [reply Sources/CodeEditTextView/STTextViewController.swift]: @lukepistrol , addressed. That code was a misguided attempt at mimicking the Xcode ruler font without having to rely on SF variable font support (which is only available in OS 13+).

**Comment by ben-p-commits** [comment]
linting issues noted. will address.

File: Sources/CodeEditTextView/STTextViewController.swift (hunk 1 above)
Comment (austincondiff) [reply Sources/CodeEditTextView/STTextViewController.swift]: @ben-p-commits As mentioned, we are relying on 13 for other things so it is safe to use variable fonts.

**Comment by lukepistrol** [comment]
Let's just wait for the review of @thecoolwinter

**Review by thecoolwinter — APPROVED** [review]
These changes look good, but I'd like to make sure we don't regress syntax highlighting in a release so let's merge this but wait to create a new release until syntax highlighting is back up and working.

**Comment by ben-p-commits** [comment]
@austincondiff , @lukepistrol - we should be all set to merge according to @thecoolwinter

**Review by austincondiff — APPROVED** [review]
Looks good!

**Comment by lukepistrol** [comment]
I'll tag a new version in a couple of hours!

**Comment by lukepistrol** [comment]
I just tagged the new release with `0.3.0`

**Pull request merged**